pkg/frontend/querymiddleware/codec.go

```diff
@@ -48,6 +48,9 @@ const (
 	statusError = "error"
 
 	totalShardsControlHeader = "Sharding-Control"
+
+	// Instant query specific options
+	instantSplitIntervalControlHeader = "Instant-Split-Interval-Control"
```

Comment:
This PR introduces specific instant query options. However, it might be useful to make this header generic in order to also debug the split interval in range queries. Let me know what you think 🙂

Reply:
As specified via slack the range query split is running in production for a long time and it was never required to control the split via HTTP header. Therefore, this will be kept only for instant queries.